pkg/lib/synthesize/generate_test.go

```diff
@@ -0,0 +1,49 @@
+package synthesize
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+const (
+	metadataDir                         = "metadata/"
+	manifestsDir                        = "manifests/"
+	validPrometheusBundlePath           = "./testdata/validBundles/prometheus_v0.22.2"
+	csvPrometheusBundleFilePath         = "./testdata/validBundles/prometheus_v0.22.2/manifests/olm.clusterserviceversion.yaml"
+	expectedCsvPrometheusBundleFilePath = "./testdata/validBundles/prometheus_v0.22.2.clusterserviceversion.yaml"
+)
+
+func TestGenerateCSV(t *testing.T) {
+	utilrand.Seed(100)
+
+	err := Generate(path.Join(validPrometheusBundlePath, metadataDir), manifestsDir, "", false)
+	require.NoError(t, err)
+	assert.EqualValues(t, loadCSV(t, expectedCsvPrometheusBundleFilePath), loadCSV(t, csvPrometheusBundleFilePath))
```

Comment:
I would prefer if we also had tests that verify that the "expected" csvPrometheusBundle is a valid bundle - this will prevent us from introducing changes that diverge csv vs csvless bundles.

Reply:
verify you mean validation or verify as in we install the CSV and see if it is successful? Validation is already done, I am working on installing CSV and see if it can get installed successfully.